tasks/install_hashi_repo.yml

```diff
@@ -65,18 +65,24 @@
         name: "{{ vault_rhsm_repo_id }}"
         state: enabled
 
+- name: Ensure Enterprise package is not present when OSS is desired and vice versa
+  package:
+    name: "{{ 'vault' ~ ('-enterprise' if not (vault_enterprise | bool) else '') }}"
```

Comment:
"if not" -> "if"?

Reply:
"not" is correct here 'cause this task is to uninstall enterprise if community is targeted and vice versa:

vault_enterprise: true -> "vault" should be absent
vault_enterprise: false -> "vault-enterprise" should be absent